Limits/python/Models.py

```diff
@@ -371,6 +376,271 @@ def build(self,ws,label):
         ws.factory("Voigtian::{0}({1}, {2}, {3}, {4})".format(label,self.x,meanName,widthName,sigmaName))
         self.params = [meanName,widthName,sigmaName]
 
+class CrystalBall(Model):
+
+    def __init__(self,name,**kwargs):
+        super(CrystalBall,self).__init__(name,**kwargs)
+
+    def build(self,ws,label):
+        logging.debug('Building {}'.format(label))
+        meanName = 'mean_{0}'.format(label)
+        sigmaName = 'sigma_{0}'.format(label)
+        aName = 'a_{0}'.format(label)
+        nName = 'n_{0}'.format(label)
+        mean  = self.kwargs.get('mean',  [1,0,1000])
+        width = self.kwargs.get('width', [1,0,100])
+        sigma = self.kwargs.get('sigma', [1,0,100])
+        a     = self.kwargs.get('a', [1,0,100])
+        n     = self.kwargs.get('n', [1,0,100])
+        # variables
+        ws.factory('{0}[{1}, {2}, {3}]'.format(meanName,*mean))
+        ws.factory('{0}[{1}, {2}, {3}]'.format(sigmaName,*sigma))
+        ws.factory('{0}[{1}, {2}, {3}]'.format(aName,*a))
+        ws.factory('{0}[{1}, {2}, {3}]'.format(nName,*n))
+        # build model
+        ws.factory("RooCBShape::{0}({1}, {2}, {3}, {4}, {5})".format(label,self.x,meanName,sigmaName,aName,nName))
+        self.params = [meanName,sigmaName,aName,nName]
+
+class CrystalBallSpline(ModelSpline):
+
+    def __init__(self,name,**kwargs):
+        super(CrystalBallSpline,self).__init__(name,**kwargs)
+
+    def build(self,ws,label):
+        logging.debug('Building {}'.format(label))
+        meanName = 'mean_{0}'.format(label)
+        sigmaName = 'sigma_{0}'.format(label)
+        aName = 'a_{0}'.format(label)
+        nName = 'n_{0}'.format(label)
+        masses = self.kwargs.get('masses', [])
+        means  = self.kwargs.get('means',  [])
+        sigmas = self.kwargs.get('sigmas', [])
+        a_s = self.kwargs.get('a_s', [])
+        n_s = self.kwargs.get('n_s', [])
+        # splines
+        meanSpline  = ROOT.RooSpline1D(meanName,  meanName,  ws.var('MH'), len(masses), array('d',masses), array('d',means))
+        sigmaSpline = ROOT.RooSpline1D(sigmaName, sigmaName, ws.var('MH'), len(masses), array('d',masses), array('d',sigmas))
+        aSpline =     ROOT.RooSpline1D(aName, aName, ws.var('MH'), len(masses), array('d',masses), array('d',a_s))
+        nSpline =     ROOT.RooSpline1D(nName, nName, ws.var('MH'), len(masses), array('d',masses), array('d',n_s))
+        # import
+        getattr(ws, "import")(meanSpline, ROOT.RooFit.RecycleConflictNodes())
```

Comment:
You could change this to use `self.wsimport(ws,...)` now that it has been added/